### DISCUSSION TITLE
Updating from older versions of Activity Browser

### DISCUSSION THREAD

**Discussion by marc-vdm** [LCA-ActivityBrowser/activity-browser#1049] [General]
# Versions below `2.10.0`
Since version `2.10.0`, we have stopped supporting python version `3.8` and `3.9`. 

You can check which version of python your environment is using with: `conda activate ab` and then `conda list python`. If your python version is `3.10` or higher, you can [update your Activity Browser like normal](https://github.com/LCA-ActivityBrowser/activity-browser/wiki/Installation-Guide#updating-activity-browser). If your python version is `3.8` or `3.9`, you need to re-install your environment following the instructions below:

> [!NOTE]  
> Don't worry about your projects and databases, that is stored safely and will still be available after following this guide.

1. Make sure you are in your `base` channel (so you have not done `conda activate ab`) with:
`conda deactivate`

2. You now need to re-create the environment with:
`conda create -n ab -c conda-forge --solver libmamba activity-browser` (of course, you can use something different from `ab` if you wish). This will completely re-create your environment, and warn you if you want to over-write the old environment, just confirm.

Now you can use AB as normal again with `conda activate ab` and `activity-browser`

When you install updates again, you can follow the normal update instructions: 
`conda update activity-browser`

# Versions below `2.7.5` or `dev 2022.xx.xx`
Ever since version `2.7.5` (or `dev 2022.xx.xx`), we have moved on from the `development` and `stable` versions to just one version, and now publish that through [`conda-forge`](https://anaconda.org/conda-forge/activity-browser).

If you have activity-browser `2.7.5` (or `dev 2022.xx.xx`) or older, you have to make some changes to your conda configuration to update to our newest version.

> [!NOTE]  
> Don't worry about your projects and databases, that is stored safely and will still be available after following this guide.

1. Make sure you are in your `base` channel (so you have not done `conda activate ab`) with:
`conda deactivate`

2. You now need to review your available channels:
`conda config --show channels`

3. Then, remove the channels `bsteubing`, `haasad`, `cmutel` and `pascallesage` if the are present in the list with:
`conda config --remove channels bsteubing` for each channel 
 
 > [!TIP]
 > If you use arrow up, you re-enter the last command in your terminal, and only need to change the name of the channel.

4. Now, you can re-create activity-browser environment, and it should install from the correct channel (`conda-forge`):
`conda create -n ab -c conda-forge --solver libmamba activity-browser` (of course, you can use something different from `ab` if you wish)
This will completely re-create your environment, and warn you if you want to over-write the old environment, just confirm.

Now you can use AB as normal again with `conda activate ab` and `activity-browser`

When you install updates again, you can follow the normal update instructions: 
`conda update activity-browser`

**Comment by zeeshankhan2** [comment]
Hi, I installed a new Ab using the instructions above, but when I want to open Ab, it does nothing and remains only in the environment. Could you please tell how to solve the issue.?
![image](https://github.com/user-attachments/assets/43733b06-a82a-49fb-a24a-dcd0b6f89f7e)

**Reply by mrvisscher** [reply]
Hi @zeeshankhan2 

Sorry to hear you're having issues. Could you send us the output of the `conda list` command for the AB environment?

Kind regards,

Marin

**Comment by zeeshankhan2** [comment]
Dear Marin, Thanks for the response. please find the list of the environment.
activity-browser          2.10.2             pyhd8ed1ab_1    conda-forge
annotated-types           0.7.0              pyhd8ed1ab_0    conda-forge
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
asteval                   1.0.4              pyhd8ed1ab_0    conda-forge
astunparse                1.6.3              pyhd8ed1ab_0    conda-forge
blinker                   1.8.2              pyhd8ed1ab_0    conda-forge
brightway2                2.4.6              pyhd8ed1ab_0    conda-forge
brotli                    1.1.0                h2466b09_2    conda-forge
brotli-bin                1.1.0                h2466b09_2    conda-forge
brotli-python             1.1.0           py311hda3d55a_2    conda-forge
brotlicffi                1.1.0.0         py311hda3d55a_2    conda-forge
bw2analyzer               0.10               pyhd8ed1ab_0    conda-forge
bw2calc                   1.8.2           py311h1ea47a8_0    conda-forge
bw2data                   3.6.6              pyhd8ed1ab_0    conda-forge
bw2io                     0.8.12             pyhd8ed1ab_0    conda-forge
bw2parameters             1.1.0              pyhd8ed1ab_0    conda-forge
bw_migrations             0.2                pyhd8ed1ab_0    conda-forge
bzip2                     1.0.8                h2466b09_7    conda-forge
ca-certificates           2024.8.30            h56e8100_0    conda-forge
certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge
cffi                      1.17.1          py311he736701_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
click                     8.1.7           win_pyh7428d3b_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.3.0           py311h3257749_2    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
dill                      0.3.8              pyhd8ed1ab_0    conda-forge
docopt                    0.6.2                      py_1    conda-forge
ecoinvent_interface       2.4.1              pyhd8ed1ab_0    conda-forge
eight                     1.0.1              pyhd8ed1ab_4    conda-forge
et_xmlfile                1.1.0              pyhd8ed1ab_0    conda-forge
fasteners                 0.17.3             pyhd8ed1ab_0    conda-forge
flask                     3.0.3              pyhd8ed1ab_0    conda-forge
fonttools                 4.54.1          py311he736701_0    conda-forge
freetype                  2.12.1               hdaf720e_2    conda-forge
future                    1.0.0              pyhd8ed1ab_0    conda-forge
glib                      2.80.5               h7025463_0    conda-forge
glib-tools                2.80.5               h4394cf3_0    conda-forge
gst-plugins-base          1.24.7               hb0a98b8_0    conda-forge
gstreamer                 1.24.7               h5006eae_0    conda-forge
h2                        4.1.0              pyhd8ed1ab_0    conda-forge
hpack                     4.0.0              pyh9f0ad1d_0    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge
icu                       75.1                 he0c23c2_0    conda-forge
idna                      3.10               pyhd8ed1ab_0    conda-forge
importlib-metadata        8.5.0              pyha770c72_0    conda-forge
importlib_metadata        8.5.0                hd8ed1ab_0    conda-forge
inflate64                 1.0.0           py311he736701_2    conda-forge
intel-openmp              2024.2.1          h57928b3_1083    conda-forge
itsdangerous              2.2.0              pyhd8ed1ab_0    conda-forge
jinja2                    3.1.4              pyhd8ed1ab_0    conda-forge
kiwisolver                1.4.7           py311h3257749_0    conda-forge
krb5                      1.21.3               hdf4eb48_0    conda-forge
lcms2                     2.16                 h67d730c_0    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libblas                   3.9.0              24_win64_mkl    conda-forge
libbrotlicommon           1.1.0                h2466b09_2    conda-forge
libbrotlidec              1.1.0                h2466b09_2    conda-forge
libbrotlienc              1.1.0                h2466b09_2    conda-forge
libcblas                  3.9.0              24_win64_mkl    conda-forge
libclang13                19.1.0          default_ha5278ca_0    conda-forge
libdeflate                1.21                 h2466b09_0    conda-forge
libexpat                  2.6.3                he0c23c2_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libglib                   2.80.5               h7025463_0    conda-forge
libhwloc                  2.11.1          default_h8125262_1000    conda-forge
libiconv                  1.17                 hcfcfb64_2    conda-forge
libintl                   0.22.5               h5728263_3    conda-forge
libintl-devel             0.22.5               h5728263_3    conda-forge
libjpeg-turbo             3.0.0                hcfcfb64_1    conda-forge
liblapack                 3.9.0              24_win64_mkl    conda-forge
libogg                    1.3.5                h2466b09_0    conda-forge
libpng                    1.6.44               h3ca93ac_0    conda-forge
libsqlite                 3.46.1               h2466b09_0    conda-forge
libtiff                   4.7.0                hb151862_0    conda-forge
libvorbis                 1.3.7                h0e60522_0    conda-forge
libwebp                   1.4.0                h2466b09_0    conda-forge
libwebp-base              1.4.0                hcfcfb64_0    conda-forge
libxcb                    1.16                 h013a479_1    conda-forge
libxml2                   2.12.7               h0f24e4e_4    conda-forge
libxslt                   1.1.39               h3df6e99_0    conda-forge
libzlib                   1.3.1                h2466b09_1    conda-forge
lxml                      4.9.3           py311h750ae06_3    conda-forge
lxmlh                     1.3.3              pyhd8ed1ab_0    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markupsafe                2.1.5           py311he736701_1    conda-forge
matplotlib-base           3.9.2           py311h8f1b1e4_1    conda-forge
mkl                       2024.1.0           h66d3029_694    conda-forge
mrio_common_metadata      0.2.1              pyhd8ed1ab_0    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
multiprocess              0.70.16         py311he736701_1    conda-forge
multivolumefile           0.2.3              pyhd8ed1ab_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
networkx                  3.3                pyhd8ed1ab_1    conda-forge
numpy                     1.26.4          py311h0b4df5a_0    conda-forge
openjpeg                  2.5.2                h3d672ee_0    conda-forge
openpyxl                  3.1.5           py311ha68e1ae_1    conda-forge
openssl                   3.3.2                h2466b09_0    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandas                    2.2.3           py311hcf9f919_1    conda-forge
pathos                    0.3.2              pyhd8ed1ab_1    conda-forge
patsy                     0.5.6              pyhd8ed1ab_0    conda-forge
pcre2                     10.44                h3d7b363_2    conda-forge
peewee                    3.17.6          py311h1eea93c_1    conda-forge
pillow                    10.4.0          py311h5592be9_1    conda-forge
pint                      0.21               pyhd8ed1ab_0    conda-forge
pip                       24.2               pyh8b19718_1    conda-forge
platformdirs              4.3.6              pyhd8ed1ab_0    conda-forge
pox                       0.3.4              pyhd8ed1ab_0    conda-forge
ppft                      1.7.6.8            pyhd8ed1ab_0    conda-forge
psutil                    6.0.0           py311he736701_1    conda-forge
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                hfa6e2cd_3    conda-forge
py7zr                     0.22.0             pyhd8ed1ab_0    conda-forge
pybcj                     1.0.2           py311he736701_1    conda-forge
pycasreg                  0.1.0              pyhd8ed1ab_0    conda-forge
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pycryptodomex             3.20.0          py311he736701_1    conda-forge
pydantic                  2.9.2              pyhd8ed1ab_0    conda-forge
pydantic-core             2.23.4          py311h533ab2d_0    conda-forge
pydantic-settings         2.5.2              pyhd8ed1ab_0    conda-forge
pyecospold                3.5.3              pyhd8ed1ab_0    conda-forge
pypardiso                 0.4.6              pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.4              pyhd8ed1ab_0    conda-forge
pyperclip                 1.9.0              pyh7428d3b_0    conda-forge
pyppmd                    1.1.0           py311hda3d55a_1    conda-forge
pyprind                   2.11.2          py311h1ea47a8_1006    conda-forge
pyside2                   5.15.8          py311h3d699ce_4    conda-forge
pysocks                   1.7.1              pyh0701188_6    conda-forge
python                    3.11.10         hce54a09_1_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-dotenv             1.0.1              pyhd8ed1ab_0    conda-forge
python-tzdata             2024.2             pyhd8ed1ab_0    conda-forge
python_abi                3.11                    5_cp311    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pyxlsb                    1.0.10             pyhd8ed1ab_0    conda-forge
pyzstd                    0.16.1          py311h53056dc_1    conda-forge
qhull                     2020.2               hc790b64_5    conda-forge
qt-main                   5.15.8              h264fbc2_26    conda-forge
qt-webengine              5.15.8               h4bf5c4e_4    conda-forge
requests                  2.32.3             pyhd8ed1ab_0    conda-forge
salib                     1.5.0              pyhd8ed1ab_0    conda-forge
scipy                     1.14.1          py311hd4686c6_0    conda-forge
seaborn                   0.13.2               hd8ed1ab_2    conda-forge
seaborn-base              0.13.2             pyhd8ed1ab_2    conda-forge
setuptools                75.1.0             pyhd8ed1ab_0    conda-forge
setuptools-scm            8.1.0              pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
stats_arrays              0.7                pyhd8ed1ab_0    conda-forge
statsmodels               0.14.3          py311h0a17f05_1    conda-forge
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
tbb                       2021.13.0            hc790b64_0    conda-forge
texttable                 1.7.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h5226925_1    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tqdm                      4.66.5             pyhd8ed1ab_0    conda-forge
types-python-dateutil     2.9.0.20240906     pyhd8ed1ab_0    conda-forge
typing-extensions         4.12.2               hd8ed1ab_0    conda-forge
typing_extensions         4.12.2             pyha770c72_0    conda-forge
tzdata                    2024a                h8827d51_1    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
unicodecsv                0.14.1             pyhd8ed1ab_2    conda-forge
unidecode                 1.3.8              pyhd8ed1ab_0    conda-forge
urllib3                   2.2.3              pyhd8ed1ab_0    conda-forge
vc                        14.3                h8a93ad2_21    conda-forge
vc14_runtime              14.40.33810         ha82c5b3_21    conda-forge
voluptuous                0.15.2             pyhd8ed1ab_1    conda-forge
vs2015_runtime            14.40.33810         h3bf8584_21    conda-forge
werkzeug                  3.0.4              pyhd8ed1ab_0    conda-forge
wheel                     0.44.0             pyhd8ed1ab_0    conda-forge
whoosh                    2.7.4           py311h1ea47a8_9    conda-forge
win_inet_pton             1.1.0              pyhd8ed1ab_6    conda-forge
wrapt                     1.16.0          py311he736701_1    conda-forge
xlrd                      2.0.1              pyhd8ed1ab_3    conda-forge
xlsxwriter                3.2.0              pyhd8ed1ab_0    conda-forge
xorg-kbproto              1.0.7             hcd874cb_1002    conda-forge
xorg-libx11               1.8.9                h0076a8d_1    conda-forge
xorg-libxau               1.0.11               hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xorg-xextproto            7.3.0             hcd874cb_1003    conda-forge
xorg-xproto               7.0.31            hcd874cb_1007    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
zipfile-deflate64         0.2.0           py311ha68e1ae_4    conda-forge
zipp                      3.20.2             pyhd8ed1ab_0    conda-forge
zstandard                 0.23.0          py311h53056dc_1    conda-forge
zstd                      1.5.6                h0ea2cb4_0    conda-forge

**Reply by mrvisscher** [reply]
I've created an issue for you: https://github.com/LCA-ActivityBrowser/activity-browser/issues/1378 let's continue the conversation there

**Comment by zeeshankhan2** [comment]
Ok
